hs-abci-examples/nameservice/src/Nameservice/Server.hs

```diff
@@ -6,29 +6,36 @@ import           Nameservice.Application                       (AppConfig (..),
                                                                 handlersContext)
 import           Network.ABCI.Server                           (serveApp)
 import           Network.ABCI.Server.App                       (Middleware)
+-- import qualified Network.ABCI.Server.Middleware.MetricsLogger  as MetLogger
 import qualified Network.ABCI.Server.Middleware.RequestLogger  as ReqLogger
 import qualified Network.ABCI.Server.Middleware.ResponseLogger as ResLogger
 import           Polysemy                                      (Sem)
 import           Tendermint.SDK.Application                    (createIOApp,
                                                                 makeApp)
-import           Tendermint.SDK.BaseApp                        (CoreEffs,
+import           Tendermint.SDK.BaseApp                        (Context (..),
+                                                                CoreEffs,
                                                                 runCoreEffs)
+import           Tendermint.SDK.BaseApp.Metrics.Prometheus     (runMetricsServer)
 
 makeAndServeApplication :: AppConfig -> IO ()
-makeAndServeApplication cfg = do
+makeAndServeApplication AppConfig{..} = do
+  _ <- runMetricsServer $ contextMetricsState baseAppContext
   putStrLn "Starting ABCI application..."
   let nat :: forall a. Sem CoreEffs a -> IO a
-      nat = runCoreEffs $ baseAppContext cfg
+      nat = runCoreEffs baseAppContext
       application = createIOApp nat $ makeApp handlersContext
   serveApp =<< hookInMiddleware application
   where
     mkMiddleware :: IO (Middleware IO)
     mkMiddleware = do
       reqLogger <- ReqLogger.mkLogStdoutDev
       resLogger <- ResLogger.mkLogStdoutDev
+      -- metLogger <- MetLogger.mkMetricsLogDatadogLocal 10516 -- local port for log collection (TDP)
+      --metLogger <- MetLogger.mkMetricsLogDatadog -- direct datadog
```

Comment:
Perhaps a local scribe is fine?

Reply:
Wouldn't you rather use environment variables so we could publish to the same datadog account that the metrics dsl is using?